03-rnaseq/ortholog-mapping_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,410 @@
+This is okay because we do not expect everything to map across species.
+
+## Take a look at some multi-mappings
+
+If a mouse Ensembl gene ID maps to multiple human Ensembl IDs, the gene expression values get duplicated.
+Let's look at the `ENSMUSG00000000290` example below.
+
+```{r}
+human_mouse_mapped_df %>%
+  dplyr::filter(Gene == "ENSMUSG00000000290")
+```
+
+Remember that our original data is associated to mouse Ensembl IDs, so when a human Ensembl ID maps to multiple mouse Ensembl IDs, the data will not get duplicated, but you will have multiple rows associated with that human symbol.
+Let's look at the `CACNA1B` example below.
+
+```{r}
+human_mouse_mapped_df %>%
+  dplyr::filter(human_ensembl == "CACNA1B")
+```
+
+## Collapse multi-mapped human Ensembl IDs
+
+Remember that if a mouse Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+We can collapse the multi-mapped values into a list for each Ensembl ID as to not have duplicate expression values in our data frame.
+
+In the next chunk, we show how we can collapse all the human Ensembl IDs into one column where we store them all for each unique mouse Ensembl ID.
+
+```{r}
+collapsed_human_ensembl_df <- human_mouse_mapped_df %>%
+  # Group by Ensembl IDs
+  dplyr::group_by(Gene) %>%
+  # Collapse the mapped values in `human_mouse_mapped_df` into one column named `all_human_ensembls`
+  dplyr::summarize(all_human_ensembls = paste(human_ensembl, collapse = ";")) %>%
+  # Now we need get the gene expression data back
+  dplyr::left_join(df, by = "Gene")
+
+head(collapsed_human_ensembl_df)
+```
+
+### Write results to file
+
+Now let's write our results to file.
+
+```{r}
+readr::write_tsv(
+  collapsed_human_ensembl_df,
+  file.path(
+    results_dir,
+    "SRP070849_mouse_ensembl_to_collapsed_human_gene_symbol.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+## Collapse multi-mapped mouse Ensembl IDs
+
+Since multiple mouse Ensembl gene IDs map to the same human symbol, we can also collapse all Ensembl IDs that each unique human Ensembl ID mapped to.
+
+A known issue in ortholog mapping is how to handle this data because combining values across mouse transcripts using an average may result in the loss of a lot of data and will likely not be representative of the mouse biology.
+Therefore, we will not add the expression data here since the expression values are not the same across Ensembl IDs, despite mapping to the same human Ensembl ID.
+
+Instead, we will use the `support` column to decide which mappings to retain.
+As we noted earlier, the prediction with the most number of databases which concur is considered the most reliable.
+Therefore, where we have multi-mapped mouse Ensembl gene IDs, we will take the mapping with the most databases that concur according to the `support` column.
+
+```{r}
+filtered_mouse_ensembl_df <- human_mouse_mapped_df %>%
```

Comment:
I would split this up a bit (show them a preview of the support column) and print out things more so we can show what's happening at each step.